### PR TITLE
Process and System memory statistics API

### PR DESCRIPTION
**Pull request by paulcbetts**
This PR adds a new API for process and system statistics, `processStats`:

```js
import {processStats} from 'electron';

processStats.getProcessMemoryInfo();
>>> Object {workingSetSize: 86052, peakWorkingSetSize: 92244, privateBytes: 25932, sharedBytes: 60248}
```

### CODE REVIEW

**Comment by zcbenz** [comment]
The functionality provided in this module overlaps with Node's builtin modules, we can probably extend Node's APIs instead of adding a new module.

For getting the process memory, I think we can merge the results to `process.memoryUsage()`, or add a new API to `process` object like `process.memoryUsageEx()`. And for getting the system memory, the `os` module already has `os.freemem()` and `os.totalmem()`, we can probably add `os.freeswapmem()` and `os.totalswap()` if you really need them.

**Comment by paulcbetts** [comment]
I can do that, though I thought we were trying to not add any new APIs to node.js?

**Comment by zcbenz** [comment]
We already have lots of extensions to the `process` object, we never extended other modules though:
https://github.com/electron/electron/blob/master/docs/api/process.md

I don't like adding APIs to Node.js, but the APIs here are perfectly suitable to be added to the built-in modules of Node.js, and it feels weird to put them elsewhere.

**Comment by paulcbetts** [comment]
@zcbenz :cool:, just wanted to grok what's what. Lemme compare how accurate the node APIs are vs Chromium (the latter seem pretty correctly implemented) then add in pieces as needed.

File: filenames.gypi [hunk 1]
```diff
@@ -292,6 +293,7 @@
       'atom/common/api/atom_api_native_image.cc',
       'atom/common/api/atom_api_native_image.h',
       'atom/common/api/atom_api_native_image_mac.mm',
+      'atom/common/api/atom_api_process_stats.cc',
```
Comment (zcbenz) [inline filenames.gypi]: Electron's extensions to `process` object are implemented in `atom/common/api/atom_bindings.cc`, so the code can be moved there.

File: docs/api/process.md [hunk 2]
```diff
@@ -14,9 +14,13 @@ upstream node:
   other builds it is `undefined`.
 * `process.windowsStore` Boolean - If the app is running as a Windows Store app
   (appx), this value is `true`, for other builds it is `undefined`.
-* `process.defaultApp` Boolean - When app is started by being passed as parameter
-  to the default app, this value is `true` in the main process, otherwise it is
-  `undefined`.
+* `process.defaultApp` Boolean - When app is started by being passed as
+  parameter to the default app, this value is `true` in the main process,
+  otherwise it is `undefined`.
+* `process.getSystemMemoryInfo` - Returns the amount of system memory and free
```
Comment (zcbenz) [inline docs/api/process.md]: We don't need to put them here since they are already listed in the Methods.

File: filenames.gypi (hunk 1 above)
Comment (paulcbetts) [inline filenames.gypi]: :+1:

**Comment by zcbenz** [comment]
👍

**Pull request merged**